core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -131,16 +132,18 @@
         this.useHttp1Pipelining = useHttp1Pipelining;
         this.idleTimeoutMillis = idleTimeoutMillis;
         this.pingIntervalMillis = pingIntervalMillis;
+        this.desiredProtocol = desiredProtocol;
+    }
 
-        switch (proxyConfig.proxyType()) {
+    private static boolean useProxyConnection(ChannelHandlerContext ctx) {
+        final PoolKey poolKey = ctx.channel().attr(POOL_KEY).get();
```

Comment:
Question: Could we pass the `poolKey` to the constructor of this class?

Reply:
Good point 👍  Modified to use `poolKey` directly for (slightly) better performance